src/centralized_telescope/telescope.rs

```diff
@@ -130,15 +131,16 @@ impl Telescope {
     ///
     /// ```
     /// use alba::centralized_telescope::Telescope;
+    /// use sha2::Sha256;
     /// let set_size = 200;
     /// let telescope = Telescope::create(64.0, 64.0, set_size, 100);
     /// let mut prover_set = Vec::new();
     /// for i in 0..set_size {
     ///     prover_set.push([(i % 256) as u8 ; 48]);
     /// }
-    /// let proof = telescope.prove(&prover_set).unwrap();
+    /// let proof = telescope.prove::<Sha256>(&prover_set).unwrap();
     /// ```
-    pub fn prove(&self, prover_set: &[Element]) -> Option<Proof> {
+    pub fn prove<H: Digest + FixedOutput>(&self, prover_set: &[Element]) -> Option<Proof<H>> {
```

Comment:
Thinking about this PR again, having domain separation in mind per design document. Two points.

1. I think we and the user would benefit from an example showing how to construct and use a hash function that has a domain id inside. Personally, right now I don't understand how easy it is to do.
2. The current interfaces would work if the user has a finite number of domain ids since they can be defined directly in the hash function type. However, if we don't know at compile time how many possible domain ids there are, the current interface will not work. The user would need to provide not just the hash function type but also a hash function object.

Reply:
I had a go at this and it is not simple indeed (as you need to deal with `OutputSizeUser` and GenericArrays).
As this is not a strict requirement for this PR, I'll open an issue to provide a hash example.